src/sign/signature.rs

```diff
@@ -1371,6 +1371,79 @@ mod test {
         }
     }
 
+    #[should_panic]
+    #[test]
+    fn signer_index_should_match_public_key() {
+        let params = ThresholdParameters::<Secp256k1Sha256>::new(2, 1);
+        let message = b"This is a test of the tsunami alert system. This is only a test.";
+
+        let p1_sk = IndividualSigningKey {
+            index: 1,
+            key: Fr::rand(&mut OsRng),
+        };
+
+        let (p1_public_comshares, _) =
+            generate_commitment_share_lists::<Secp256k1Sha256>(&mut OsRng, &p1_sk, 1).unwrap();
+
+        let mut aggregator = SignatureAggregator::new(
+            params,
+            GroupVerifyingKey::new(Projective::zero()),
+            &message[..],
+        );
+
+        // The aggregator should panic as the participant index does not match the public key index.
+        aggregator.include_signer(3, p1_public_comshares.commitments[0], &p1_sk.to_public());
+    }
+
+    #[should_panic]
+    #[test]
+    fn aggregator_get_too_many_signers() {
+        let params = ThresholdParameters::<Secp256k1Sha256>::new(3, 2);
+        let message = b"This is a test of the tsunami alert system. This is only a test.";
+
+        let p1_sk = IndividualSigningKey {
+            index: 1,
+            key: Fr::rand(&mut OsRng),
+        };
+        let p2_sk = IndividualSigningKey {
+            index: 2,
+            key: Fr::rand(&mut OsRng),
+        };
+        let p3_sk = IndividualSigningKey {
+            index: 3,
+            key: Fr::rand(&mut OsRng),
+        };
+        let p4_sk = IndividualSigningKey {
+            index: 4,
+            key: Fr::rand(&mut OsRng),
+        };
+
+        let (p1_public_comshares, _) =
+            generate_commitment_share_lists::<Secp256k1Sha256>(&mut OsRng, &p1_sk, 1).unwrap();
```

Comment:
I was confused because the name of this method doesn't mention the nonces. What about `generate_nonce_commitment_share_lists` (at least the paper says "nonce/commitment share'). Still, I don't see why the paper talks about "share" when you only have nonces (e_{ij}, d_{i,j}) and commitments to this nonces (D_{ij}, E_{ij})

Reply:
And unrelated, why is E hidding? For preventing some attack?